packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -103,9 +103,12 @@ class LandmarkManager {
     this.landmarks.splice(insertPosition, 0, newLandmark);
   }
 
-  public updateLandmark(landmark: Landmark) {
-    this.landmarks = this.landmarks.map((prevLandmark) => prevLandmark.ref.current === landmark.ref.current ? {...prevLandmark, ...landmark} : prevLandmark);
-    this.checkLabels(landmark.role);
+  public updateLandmark(ref: MutableRefObject<HTMLElement>, newFields: Partial<Omit<Landmark, 'ref'>>) {
+    let index = this.landmarks.findIndex(l => l.ref === ref);
+    if (index >= 0) {
+      this.landmarks[index] = {...this.landmarks[index], ...newFields};
```

Comment:
given that `l.ref` and `landmark.ref` are the same, we can just spread `landmark` here if we go back to passing landmark in as the single arg